src/txn/txn_timestamp.c

```diff
@@ -524,14 +536,13 @@ __wt_txn_validate_commit_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *com
               __wt_timestamp_to_string(stable_ts, ts_string[1]));
 
         /*
-         * Compare against the commit timestamp of the current transaction. Return an error if the
-         * given timestamp is older than the first commit timestamp.
+         * For non-prepared transactions the first commit timestamp must be after the stable
+         * timestamp.
          */
-        if (F_ISSET(txn, WT_TXN_HAS_TS_COMMIT) && commit_ts < txn->first_commit_timestamp)
-            WT_RET_MSG(session, EINVAL,
-              "commit timestamp %s older than the first commit timestamp %s for this transaction",
+        if (F_ISSET(txn, WT_TXN_HAS_TS_COMMIT) && commit_ts <= stable_ts)
+            WT_RET_MSG(session, EINVAL, "commit timestamp %s must be after the stable timestamp %s",
               __wt_timestamp_to_string(commit_ts, ts_string[0]),
-              __wt_timestamp_to_string(txn->first_commit_timestamp, ts_string[1]));
+              __wt_timestamp_to_string(stable_ts, ts_string[1]));
```

Comment:
Is this test the same as the one immediately above, `if (has_stable_ts && commit_ts <= stable_ts)`?

Reply:
That is correct, my bad! Removed in 89d5815a8e472a0f2a49565deee7fe5608ff2398